src/main/java/edu/cornell/kfs/sys/aws/batch/AwsSecretValidationStep.java

```diff
@@ -0,0 +1,134 @@
+            awsSecretService.logCacheStatus();
+
+        } catch (JsonMappingException e) {
+            LOG.error("execute, had an error mapping AWS Secrets to POJO", e);
+            throw new RuntimeException(e);
+        } catch (JsonProcessingException e) {
+            LOG.error("execute, had an error process AWS Secrets", e);
+            throw new RuntimeException(e);
+        }
+        
+        validateUsingNewAWSServiceFromSpring();
+
+        awsSecretService.clearCache();
+        LOG.info("execute, ending job");
+        return true;
+    }
+
+    private Map<String, Object> buildNewSecretValue() {
+        Map<String, Object> values = new HashMap<String, Object>();
+        values.put(ACCESS_TOKEN, UUID.randomUUID().toString());
+        values.put(REFRESH_TOKEN, UUID.randomUUID().toString());
+        values.put(ACCESS_TOKEN_EXPIRATION_DATE, Calendar.getInstance(Locale.US).getTime());
+        return values;
+    }
+    
+    protected void updateAmazonSecretValidationInstance(AmazonSecretValidationInstance instance,
+            Map<String, Object> newSecretValues) {
+        instance.setAccess_token(newSecretValues.get(ACCESS_TOKEN).toString());
+        instance.setRefresh_token(newSecretValues.get(REFRESH_TOKEN).toString());
+        instance.setAccess_token_expiration_date((Date) newSecretValues.get(ACCESS_TOKEN_EXPIRATION_DATE));
+    }
+
+    protected void confirmInstanceSecretValues(Map<String, Object> expectedSecretValues,
+            AmazonSecretValidationInstance instanceValues) {
+        String errorMessage = StringUtils.EMPTY;
+        if (!StringUtils.equals(instanceValues.getAccess_token(),
+                expectedSecretValues.get(ACCESS_TOKEN).toString())) {
+            errorMessage = errorMessage + "Did not update ACCESS token as expected.";
+        }
+        if (!StringUtils.equals(instanceValues.getRefresh_token(),
+                expectedSecretValues.get(REFRESH_TOKEN).toString())) {
+            errorMessage = errorMessage + " Did not update REFRESH token as expected.";
+        }
+        if (!instanceValues.getAccess_token_expiration_date()
+                .equals((Date) expectedSecretValues.get(ACCESS_TOKEN_EXPIRATION_DATE))) {
+            errorMessage = errorMessage + " Did not update access token EXPIRATION DATE as expected.";
+        }
+
+        if (StringUtils.isNotBlank(errorMessage)) {
+            throw new RuntimeException(errorMessage);
+        }
+    }
+    
+    protected void validateUsingNewAWSServiceFromSpring() {
+        AwsSecretService service = SpringContext.getBean(AwsSecretService.class);
+        try {
+            AmazonSecretValidationShared sharedSpaceSecrets = service.getPojoFromAwsSecret(AWS_SECRET_NAME_VALIDATION_SHARED, 
+                    false, AmazonSecretValidationShared.class);
+            LOG.info("validateUsingNewAWSServiceFromSpring, sharedSpaceSecrets: " + sharedSpaceSecrets);
```

Comment:
Does this log statement add any value, now that the DTO's custom toString() method has been removed?

Reply:
replaced with the function confirming values of AmazonSecretValidationShared